cmd/docker/docker.go

```diff
@@ -325,6 +326,25 @@ func tryPluginRun(ctx context.Context, dockerCli command.Cli, cmd *cobra.Command
 	return nil
 }
 
+// registerForceExitGoroutine registers a goroutine that will force exit the
+// process after 3 SIGTERM/SIGINT signals.
+func registerForceExitGoroutine(ctx context.Context, w io.Writer) {
+	// setup a signal handler to force exit after 3 SIGTERM/SIGINT
+	go func() {
```

Comment:
Synchronous functions should be preferred over the asynchronous. Especially when the whole function body is wrapped in the goroutine - it's better to leave the `go ` up to the caller.

Reply:
I would say that the function quite explicitly mentions that it's a background task and this is not an uncommon pattern in Go. It's also not exported so it should only be used inside this package - and only once. I would use a closure, but thought wrapping it in a function with a name that's explicit to its intent would make it more obvious to future contributors.

Reply:
Encoding the behavior with language feature is always more readable and explicit than using a function name.

There's no need to use a closure on the caller side:
```go
go forceExitAfter3TerminationsSignals(ctx, dockerCli.Err())
```

Reply:
Alright, I've updated it accordingly